[PATCH] dontaudit several operations of useradd, newrole, and ssh

passwd, newrole and semanage need not to search init_var_run_t
systemd_logind needs not to mounton var_auth and read tmp_t
semanage needs not to write lib_t

Upstream-Status: Pending

Signed-off-by: Roy Li <rongqing.li@windriver.com>
Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/admin/usermanage.te   |  2 ++
 policy/modules/system/authlogin.if   | 19 +++++++++++++++++++
 policy/modules/system/init.if        | 19 +++++++++++++++++++
 policy/modules/system/selinuxutil.te |  5 +++++
 policy/modules/system/systemd.te     |  3 +++
 5 files changed, 48 insertions(+)

--- refpolicy.orig/policy/modules/admin/usermanage.te
+++ refpolicy/policy/modules/admin/usermanage.te
@@ -358,6 +358,8 @@ userdom_dontaudit_search_user_home_conte
 
 mls_file_read_all_levels(passwd_t)
 
+init_dontaudit_search_pid_dirs(passwd_t)
+
 optional_policy(`
 	nscd_run(passwd_t, passwd_roles)
 ')
--- refpolicy.orig/policy/modules/system/authlogin.if
+++ refpolicy/policy/modules/system/authlogin.if
@@ -1011,6 +1011,25 @@ interface(`auth_manage_var_auth',`
 	allow $1 var_auth_t:lnk_file rw_lnk_file_perms;
 ')
 
+
+########################################
+## <summary>
+## </summary>
+## <param name="type">
+##	<summary>
+##	</summary>
+## </param>
+#
+interface(`auth_dontaudit_mounton_var_auth',`
+	gen_require(`
+		type var_auth_t;
+	')
+
+	dontaudit $1 var_auth_t:dir mounton;
+')
+
+
+
 ########################################
 ## <summary>
 ##	Read PAM PID files.
--- refpolicy.orig/policy/modules/system/init.if
+++ refpolicy/policy/modules/system/init.if
@@ -2200,6 +2200,25 @@ interface(`init_getattr_utmp',`
 	allow $1 initrc_var_run_t:file getattr;
 ')
 
+######################################
+## <summary>
+##  dontaudit search  directory in the /run/systemd directory.
+## </summary>
+## <param name="domain">
+##  <summary>
+##  Domain allowed access.
+##  </summary>
+## </param>
+#
+interface(`init_dontaudit_search_pid_dirs',`
+	gen_require(`
+		type init_var_run_t;
+	')
+
+	dontaudit $1 init_var_run_t:dir search_dir_perms;
+')
+
+
 ########################################
 ## <summary>
 ##	Read utmp.
--- refpolicy.orig/policy/modules/system/selinuxutil.te
+++ refpolicy/policy/modules/system/selinuxutil.te
@@ -291,6 +291,8 @@ miscfiles_read_localization(newrole_t)
 
 seutil_libselinux_linked(newrole_t)
 
+init_dontaudit_search_pid_dirs(newrole_t)
+
 # for some PAM modules and for cwd
 userdom_dontaudit_search_user_home_content(newrole_t)
 userdom_search_user_home_dirs(newrole_t)
@@ -504,6 +506,9 @@ seutil_get_semanage_read_lock(semanage_t
 # netfilter_contexts:
 seutil_manage_default_contexts(semanage_t)
 
+init_dontaudit_search_pid_dirs(semanage_t)
+libs_dontaudit_write_lib_dirs(semanage_t)
+seutil_sigchld_newrole(semanage_t)
 # Handle pp files created in homedir and /tmp
 userdom_read_user_home_content_files(semanage_t)
 userdom_read_user_tmp_files(semanage_t)
--- refpolicy.orig/policy/modules/system/systemd.te
+++ refpolicy/policy/modules/system/systemd.te
@@ -328,6 +328,9 @@ domain_read_all_domains_state(systemd_lo
 userdom_manage_all_user_tmp_content(systemd_logind_t)
 userdom_mounton_tmp_dirs(systemd_logind_t)
 
+files_dontaudit_list_non_security(systemd_logind_t)
+auth_dontaudit_mounton_var_auth(systemd_logind_t)
+
 optional_policy(`
 	dbus_system_bus_client(systemd_logind_t)
 	dbus_connect_system_bus(systemd_logind_t)
